### DISCUSSION TITLE
Is there any way to convert/export the FilmonTV streams to a m3u/m3u8 format?

### DISCUSSION THREAD

**Discussion by ANVIL88o** [streamlink/streamlink#3840] [Q&A] [answered]
Is there any way to convert/export the FilmonTV streams to a m3u/m3u8 format? So it can be used on an IPTV player app on a smart TV?

**Comment by bastimeyer** [comment, marked as answer]
The resolved HLS stream URL(s) can always be printed to stdout via `--stream-url` or `--json`, which you can then use in a different application as input. Sometimes you'll have to set additional HTTP headers though. See `--json` for all the headers which Streamlink sets.

The `FilmOn` plugin however requires a custom HLS implementation, so I'm not sure if it's possible to return the resolved HLS URL and use it in another application without any issues.
https://github.com/streamlink/streamlink/blob/1dce48708d6a4ca0d6f2f50924a8d0cdf2391a13/src/streamlink/plugins/filmon.py#L73

A workaround for that is piping Streamlink's stream output into ffmpeg and serve the HLS stream this way. Take a look at the ffmpeg manual for the available parameters and configs.
